src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/PrimitiveDataContract.cs

```diff
@@ -159,24 +174,24 @@ internal CharDataContract(XmlDictionaryString name, XmlDictionaryString ns) : ba
         {
         }
 
-        internal override string WriteMethodName { get { return "WriteChar"; } }
-        internal override string ReadMethodName { get { return "ReadElementContentAsChar"; } }
+        internal override string WriteMethodName => "WriteChar";
+        internal override string ReadMethodName => "ReadElementContentAsChar";
```

Comment:
I think you might be able to use `nameof(XmlWriterDelegator.WriteChar)`

Reply:
These are the name of a method on both `XmlWriterDelegator` and `XmlObjectSerializerWriteContext`, which are not officially related.  These are all internal, so I'm sure there is a better way to hook this all up without hoping that we don't have a typo in a string or that a method "pointed to" by a string doesn't go missing... but I'd rather save that for a later PR. Doing it here using `nameof` on only one of the types is only half a fix.